haproxy_discovery.sh

```diff
@@ -14,6 +14,10 @@
 HAPROXY_SOCK="/var/run/haproxy/info.sock"
 [ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
 
+# Define variables and perform checks
+CONFIG_FILE="$(dirname $(readlink -f "$0"))/haproxy_config"
+. $CONFIG_FILE
+
```

Comment:
As far as location of config file goes.
I'd prefer `/etc/zabbix/zabbix-haproxy.conf` or similar..
perhaps, even `/etc/zabbix-haproxy.conf` or `/etc/default/zabbix-haproxy`